### DISCUSSION TITLE
How to use the output of a process to check for a condition?

### DISCUSSION THREAD

**Discussion by saberiato** [nextflow-io/nextflow#4287] [Q&A] [answered]
Hi,
I can't find a way to use the output of a process in a condition...
Consider we have this process:
```
process myTask {
    output:
    env FOO, emit: FOO

    script:
    '''
    FOO=0
    '''
}
```

And, I want to check the output `FOO` in a condition:
```
if(myTask.out.FOO == 0) {
...
}
```
But, It doesn't work... I tried `toInteger()`, `toList()[0]`, `first()`, ..., but couldn't find a way to do that...

Also, I need to check if the output of a process is an empty file in a condition again... but no luck...

I appreciate any help!

**Comment by mribeirodantas** [comment, marked as answer]
The appropriate way to handle channels is to use channel operators. You can use the `branch` operator (docs [here](https://www.nextflow.io/docs/latest/operator.html#branch)) for checking a condition. It will create N new channels, where N is the number of conditions that you described, distributing the channel elements according to your condition. If you want something more elaborate, you can also use a Nextflow process. If you want a process to only create tasks based on a condition, you can also use the `when` block (docs [here](https://www.nextflow.io/docs/latest/process.html#when)).

The rule of thumb to remember: If you want to do something with a channel, you need to use a channel operator (or a process) 😉

**Reply by saberiato** [reply]
Thanks @mribeirodantas.
I used `branch` operator, and it worked.
